pkg/apis/kubermatic/v1/external_cluster.go

```diff
@@ -64,6 +64,125 @@ type ExternalClusterSpec struct {
 
 	KubeconfigReference *providerconfig.GlobalSecretKeySelector `json:"kubeconfigReference,omitempty"`
 	CloudSpec           *ExternalClusterCloudSpec               `json:"cloudSpec,omitempty"`
+	KubeOneSpec         *ExternalClusterKubeOneSpec             `json:"kubeoneSpec,omitempty"`
+}
+
+type ExternalClusterKubeOneSpec struct {
+	SSHReference      *providerconfig.GlobalSecretKeySelector `json:"sshReference,omitempty"`
+	ManifestReference *providerconfig.GlobalSecretKeySelector `json:"manifestReference,omitempty"`
+	CloudSpec         *KubeOneCloudSpec                       `json:"cloudSpec,omitempty"`
+}
+
+type KubeOneCloudSpec struct {
+	AWS          *KubeOneAWSCloudSpec          `json:"aws,omitempty"`
+	GCP          *KubeOneGCPCloudSpec          `json:"gcp,omitempty"`
+	Azure        *KubeOneAzureCloudSpec        `json:"azure,omitempty"`
+	Digitalocean *KubeOneDigitaloceanCloudSpec `json:"digitalocean,omitempty"`
+	Openstack    *KubeOneOpenstackCloudSpec    `json:"openstack,omitempty"`
+	Packet       *KubeOnePacketCloudSpec       `json:"packet,omitempty"`
+	Hetzner      *KubeOneHetznerCloudSpec      `json:"hetzner,omitempty"`
+	VSphere      *KubeOneVSphereCloudSpec      `json:"vsphere,omitempty"`
+	Nutanix      *KubeOneNutanixCloudSpec      `json:"nutanix,omitempty"`
+}
```

Comment:
What made you design the types this way, instead of considering KubeOne to be just another provider like GKE/AKS? I don't think we want to (or even can) distill KubeOne clusters down to "oh yeah this kubeone cluster runs on vsphere".

Reply:
Yeah, simply saying, Kubeone should be just another externalCluster, inside the `ExternalClusterCloudSpec`.

Reply:
"KubeOne require you to have the cloud provider credentials exported as the environment variables. "
https://docs.kubermatic.com/kubeone/v1.4/tutorials/creating_clusters/
this design is to get the required credentials -> save in secret -> export them in  kubeone pod

Reply:
I can move this kubeone here:
```// ExternalClusterCloudSpec mutually stores access data to a cloud provider.
type ExternalClusterCloudSpec struct {
	GKE         *ExternalClusterGKECloudSpec `json:"gke,omitempty"`
	EKS         *ExternalClusterEKSCloudSpec `json:"eks,omitempty"`
	AKS         *ExternalClusterAKSCloudSpec `json:"aks,omitempty"`
	KubeOneSpec *ExternalClusterKubeOneSpec  `json:"kubeone,omitempty"`
}
```

Reply:
ok